lua/lang/cpp.lua

```diff
@@ -1,24 +1,22 @@
 local M = {}
 
 M.config = function()
-  O.lang.clang = {
+  O.lang.cpp = {
     diagnostics = {
       virtual_text = { spacing = 0, prefix = "" },
       signs = true,
       underline = true,
     },
-    cross_file_rename = true,
-    header_insertion = "never",
     filetypes = { "c", "cpp", "objc" },
-    formatter = {
-      exe = "clang-format",
-      args = {},
-      stdin = true,
-    },
-    linters = {
-      "cppcheck",
-      "clangtidy",
+    clangd_flags = {
+      "--background-index",
+      "--header-insertion=never",
+      "--cross-file-rename",
+      "--clang-tidy",
+      "--clang-tidy-checks=-*,llvm-*,clang-analyzer-*,modernize-*",
```

Comment:
Why providing a bunch of enabled checks ?

Reply:
The whole point is that the user can override `O.lang.cpp.linters` if they want. I just made them empty by default.

It's some weird behaviour by `clangd` where it doesn't know how to run `clang-tidy` without some pre-defined checks. I'm only using the basic ones. 

I forgot to remove "modernize" which is useful for testing but may overwhelm the user.

Reply:
Ok :+1:  I wasn't aware of that since I use clang-format and clang-tidy separately.